zebrad/tests/common/lightwalletd/send_transaction_test.rs

```diff
@@ -209,14 +206,9 @@ pub async fn run() -> Result<()> {
         counter += 1;
     }
 
-    // This RPC has temporarily been disabled in `lightwalletd`:
-    // https://github.com/adityapk00/lightwalletd/blob/b563f765f620e38f482954cd8ff3cc6d17cf2fa7/frontend/service.go#L529-L531
-    //
-    // TODO: re-enable it when lightwalletd starts returning transactions again.
-    //assert!(counter >= 1, "all transactions from future blocks failed to send to an isolated mempool");
-    assert_eq!(
-        counter, 0,
-        "developers: update this test for lightwalletd sending transactions"
+    assert!(
+        counter >= 1,
+        "all transactions from future blocks failed to send to an isolated mempool"
     );
```

Comment:
If you don't think it's worth looking into these failures any more, we could also split the PR:
1. Merge all the working changes above this line
2. Open a separate ticket for all the failing changes in these lines and below.

I don't know why this part of the test is failing, but maybe it's ok to just fix some of it now, and work the rest out later?

It would be nice to have good coverage of sending mempool transactions, but if we've survived without it for more than a year, maybe it's not that important?

It's your decision.

Reply:
I made changes to go back to check the count of transactions obtained from the mempool using grpc is `0` as it was before. This now fails locally here but passed the CI. Maybe the fixes we are doing to the cache state will help, if so, this tests should start failing. 

If it is only a test issue (which we think is the case) then this is not very important, if transactions are not getting into mempool then this is critical.

I opened a ticket for further investigation in the future. https://github.com/ZcashFoundation/zebra/issues/7529

Reply:
> If it is only a test issue (which we think is the case) then this is not very important, if transactions are not getting into mempool then this is critical.

You're right, that woulds be serious. But as far as I can see from the Zebra logs, the transactions get into the mempool. So that part of the test passes.

`lightwalletd` just can't seem to read them back immediately in our tests. But we know there have been fixes and changes to those gRPC calls recently, including disabling and re-enabling them. So it could be a bug in the tests, a timing bug, or a specific issue with our test setup.

I'm happy to work this out in the ticket.